sklearn/_loss/loss.py

```diff
@@ -994,6 +996,78 @@ def gradient_proba(
         )
 
 
+class ExponentialLoss(BaseLoss):
+    """Exponential loss with (half) logit link, for binary classification.
+
+    This is also know as boosting loss.
+
+    Domain:
+    y_true in [0, 1], i.e. regression on the unit interval
+    y_pred in (0, 1), i.e. boundaries excluded
```

Comment:
For my education: why are 0 and 1 excluded for `y_pred`?

Reply:
Same as for log loss:
1. ypred is computed via a function that never reaches 0 nor 1 (only if +-inf is passed as argument). See the link function.
2. ypred is a probability. If ypred=ytrue, it‘s fine. But, e.g., for ypred=1 and ytrue=0, the loss is infinity (or undefined).

Therefore it‘s excluded here.

Reply:
Thanks!